examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp

```diff
@@ -93,68 +93,109 @@ CHIP_ERROR EVSEManufacturer::Shutdown()
+
+    ElectricalPowerMeasurementDelegate * dg = mn->GetEPMDelegate();
+    VerifyOrReturnError(dg != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    // Set power mode to AC
+    ReturnErrorOnFailure(dg->SetPowerMode(PowerModeEnum::kAc));
+
+    return CHIP_NO_ERROR;
+}
+
 /**
  * @brief   Allows a client application to send in power readings into the system
  *
- * @param[in]  aEndpointId     - Endpoint to send to EPM Cluster
- * @param[in]  aActivePower_mW - Power measured in milli-watts
- * @param[in]  aVoltage_mV     - Voltage measured in milli-volts
- * @param[in]  aCurrent_mA     - Current measured in milli-amps
+ * @param[in]  aEndpointId       - Endpoint to send to EPM Cluster
+ * @param[in]  aActivePower_mW   - ActivePower measured in milli-watts
+ * @param[in]  aVoltage_mV       - Voltage measured in milli-volts
+ * @param[in]  aActiveCurrent_mA - ActiveCurrent measured in milli-amps
  */
 CHIP_ERROR EVSEManufacturer::SendPowerReading(EndpointId aEndpointId, int64_t aActivePower_mW, int64_t aVoltage_mV,
-                                              int64_t aCurrent_mA)
+                                              int64_t aActiveCurrent_mA)
 {
-    // TODO add Power Readings when EPM cluster is merged
+    EVSEManufacturer * mn = GetEvseManufacturer();
+    VerifyOrReturnError(mn != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    ElectricalPowerMeasurementDelegate * dg = mn->GetEPMDelegate();
+    VerifyOrReturnError(dg != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    dg->SetActivePower(MakeNullable(aActivePower_mW));
+    dg->SetVoltage(MakeNullable(aVoltage_mV));
+    dg->SetActiveCurrent(MakeNullable(aActiveCurrent_mA));
 
     return CHIP_NO_ERROR;
 }
 
 /**
- * @brief   Allows a client application to send in energy readings into the system
+ * @brief   Allows a client application to send cumulative energy readings into the system
  *
  *          This is a helper function to add timestamps to the readings
  *
  * @param[in]  aCumulativeEnergyImported -total energy imported in milli-watthours
  * @param[in]  aCumulativeEnergyExported -total energy exported in milli-watthours
  */
-CHIP_ERROR EVSEManufacturer::SendEnergyReading(EndpointId aEndpointId, int64_t aCumulativeEnergyImported,
-                                               int64_t aCumulativeEnergyExported)
+CHIP_ERROR EVSEManufacturer::SendCumulativeEnergyReading(EndpointId aEndpointId, int64_t aCumulativeEnergyImported,
+                                                         int64_t aCumulativeEnergyExported)
 {
     MeasurementData * data = MeasurementDataForEndpoint(aEndpointId);
 
     EnergyMeasurementStruct::Type energyImported;
     EnergyMeasurementStruct::Type energyExported;
 
     // Get current timestamp
-    uint32_t currentTimestamp;
-    CHIP_ERROR err = GetEpochTS(currentTimestamp);
+    uint32_t currentTimestamp = 0;
+    uint64_t nowMS            = System::SystemClock().GetMonotonicMilliseconds64().count(); // In case we can't get real time
```

Comment:
This might not be the "system time" in the system-ms sense of the spec.

The right thing to use here is `app::server::TimeSinceInit` (but then convert it to milliseconds, of course).